.github/workflows/build.yml

```diff
@@ -340,3 +369,44 @@ jobs:
       - name: copy binaries to s3
         run: |
           aws s3 sync . s3://apex-net/
+
+  build-rpm:
+    needs: ["go-lint"]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: docker/setup-buildx-action@v2
+      - name: Install make
+        run: |
+          sudo apt -y install make
+      - name: Build rpm
+        run: |
+          make rpm
+
+  publish-rpms:
+    needs: ["build-rpm", "e2e"]
+    runs-on: ubuntu-latest
+    environment: copr-repo
+    strategy:
+      matrix:
+        include:
+          - mock_root: fedora-37-x86_64
+            srpm_distro: fc37
+          - mock_root: fedora-38-x86_64
+            srpm_distro: fc38
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: docker/setup-buildx-action@v2
+      - name: Install make
+        run: |
+          sudo apt -y install make
+      - name: Build srpm
+        run: |
+          MOCK_ROOT="${{ matrix.mock_root }}" make srpm
```

Comment:
It's been ages since I last build RPMs... but aren't there some directories that we might want/need to cache between builds to keep build times low.

Reply:
Every source RPM is built in a new chroot, because each one is a different distro/version.

The only case I know of that could benefit is the `build-rpm` job because it's building the source rpm and then a binary rpm using the same chroot type. I could reuse it instead of having it rebuilt. It doesn't take that long, and doing everything in a clean chroot is probably better.  I'll take another close look at the times to see if it's worth it.

Reply:
So far I sped things up about 10% by just telling it not to do any cleaning of the chroot before and after the build. Each run of `mock` happens in a new container anyway.

I know there's more I can do, so I'll keep poking at it over time.